### PR TITLE
Add uncompress function for deflate compress data

### PR DESCRIPTION
**Pull request by k50661184**
Add adler32 func(which is based on the adler algorithm) in the red/runtime/crypto.reds. 
Add the inflate.reds in red/runtime, which is based on the deflate algorithm(include the Lz77 and huffman trees).
Add the inflate-test.reds to test three happenings in the inflate process, because the compressed data have three happenings when it was in the deflate process(use fixed trees, use dynamic trees, umcompress func)
Add the test12.gz , this file is for the inflate-test.reds'inflate-uncompressed-block.

### CODE REVIEW

**Comment by greggirwin** [comment]
Huang, wow! Lots of great work in there.

Typo at: "File: 	%infalte-test.reds", should be "inflate" in there.

I also see `deflate/uncompress` in the commented test area, but `deflate` not defined anywhere. Likely just a temporary problem as you work.

Good to see you've started a tests file. Really important for code like this. I imagine the team will ask for some tests in QuickTest format to go with this PR.

**Comment by k50661184** [comment]
@greggirwin  I am glad to what you said.
The "inflate" problem i have solved. 
The "deflate/uncompress"  should be there because this function is based on the deflate algorithm.
And i have modified the "inflate context" to "deflate context" in the inflate.reds.
I'm a novice in coding, and thank you for these advice.
Well, i will close this pr and pr again.

**Comment by k50661184** [comment]
In this pr, i have changed the tests' action in the inflate-test.reds.
The inflate-test read the file and store it in a buffer,then read the .gz file and store it in another buffer.
if this two buffer's content are the same, it will output "true“. or,it will output "false"

File: runtime/inflate.reds [hunk 1]
```diff
@@ -0,0 +1,831 @@
...
+		until [
+			;--read final block flag
+			bfinal: getbit d
+			;--read block type (2 bits)
+			btype: read-bits d 2 0
+			switch btype [
+				0 [
+					;--decompress uncompressed block
+					res: inflate-uncompressed-block d
+					probe "decompress uncompressed block"
```
Comment (qtxie) [inline runtime/inflate.reds]: Should remove all the `probe` statement in this file.

File: runtime/inflate.reds [hunk 2]
```diff
@@ -0,0 +1,822 @@
...
+			xlen    [integer!]
+			hcrc    [integer!]
+			i       [integer!]
+			res     [integer!]
+			a       [integer!]
+			b       [integer!]
+			c       [integer!]
+			flga    [integer!]
+	][  
+		src: as byte-ptr! system/stack/allocate 1000
```
Comment (qtxie) [inline runtime/inflate.reds]: What's the meaning of this line?

File: runtime/inflate.reds [hunk 3]
```diff
@@ -0,0 +1,822 @@
...
+            dst     [byte-ptr!]
+            a32     [integer!]
+            cmf     [byte!]
+            flg     [byte!]
+            a       [integer!]
+            b       [integer!]
+            c       [integer!]
+            res     [integer!]
+    ][  
+        src: as byte-ptr! system/stack/allocate 1000
```
Comment (qtxie) [inline runtime/inflate.reds]: Why allocate some memory here?

**Comment by k50661184** [comment]
I have fixed and tested the adler32 func in the crypto.reds, the func is ok.
Deleted the allocate block in the gzip-compress and zlib-compress.

**Comment by qtxie** [comment]
@k50661184 OK. I'll implement `decompress` function in Red base on it. Thanks. 👍

**Comment by qtxie** [comment]
@k50661184 I got an error when decompress test51.gz. Would you check it again ?

**Comment by k50661184** [comment]
I have check it again, the error is not exist in my file. The inflate-uncompressed-block func is ok.

**Comment by k50661184** [comment]
Debug the getbit func in inflate.reds.  The "src" positiones in infalte-test.reds were  changed.

**Pull request merged**